src/ansi-c/padding.cpp

```diff
@@ -126,9 +126,10 @@ void add_padding(struct_typet &type, const namespacet &ns)
       else if(bit_field_bits!=0)
       {
         // not on a byte-boundary?
-        if((bit_field_bits%8)!=0)
+        if((bit_field_bits % config.ansi_c.char_width) != 0)
```

Comment:
I'm probably misunderstanding C bitfields here... But this almost looks like you are doing the opposite of what the PR claims, and making the assumption here that char_width == byte? I think this code really does need to deal with byte width rather than char width? Apologies if I've misunderstood though.

Reply:
Maybe my commit message/description is messed up, but indeed `char_width == byte` is what's guaranteed (while `byte == 8 bits` isn't):

"**byte** addressable unit of data storage large enough to hold any member of the basic character set of the execution environment" and "A byte contains **CHAR_BIT** bits, and the values of type unsigned char range from 0 to 2^CHAR_BIT − 1."

Reply:
Sorry, my mistake - I think I mis-read the title of the PR, and thanks for the clarification and putting me back on track :-)